sentinelhub/api/batch/utils.py

```diff
@@ -2,31 +2,93 @@
 Module implementing utilities for working with batch jobs.
 """
 import logging
+import sys
 import time
+import warnings
 from collections import defaultdict
-from typing import DefaultDict, List, Optional
+from typing import DefaultDict, List, Optional, TypeVar, Union, overload
 
 from tqdm.auto import tqdm
 
 from ...config import SHConfig
 from .base import BatchRequestStatus
-from .process import BatchRequest, BatchRequestType, BatchTileStatus, SentinelHubBatch
+from .process import BatchRequest, BatchTileStatus, SentinelHubBatch
+from .statistical import BatchStatisticalRequest, SentinelHubBatchStatistical
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal  # pylint: disable=ungrouped-imports
 
 LOGGER = logging.getLogger(__name__)
 
+BatchRequestSpec = Union[str, dict, BatchRequest, BatchStatisticalRequest]
+BatchRequestType = TypeVar("BatchRequestType", BatchStatisticalRequest, BatchRequest)  # pylint: disable=invalid-name
+BatchKind = Literal["process", "statistical"]
+
 
 _MIN_SLEEP_TIME = 60
 _DEFAULT_SLEEP_TIME = 120
 _MIN_ANALYSIS_SLEEP_TIME = 5
 _DEFAULT_ANALYSIS_SLEEP_TIME = 10
 
 
+@overload
 def monitor_batch_job(
-    batch_request: BatchRequestType,
+    batch_request: Union[str, dict],
     config: Optional[SHConfig],
     sleep_time: int = _DEFAULT_SLEEP_TIME,
     analysis_sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: Literal["process"],
 ) -> DefaultDict[BatchTileStatus, List[dict]]:
+    pass
+
+
+@overload
+def monitor_batch_job(
+    batch_request: BatchRequest,
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_SLEEP_TIME,
+    analysis_sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: BatchKind,
+) -> DefaultDict[BatchTileStatus, List[dict]]:
+    pass
+
+
+@overload
+def monitor_batch_job(
+    batch_request: Union[str, dict],
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_SLEEP_TIME,
+    analysis_sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: Literal["statistical"],
+) -> None:
+    pass
+
+
+@overload
+def monitor_batch_job(
+    batch_request: BatchStatisticalRequest,
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_SLEEP_TIME,
+    analysis_sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: BatchKind,
+) -> None:
+    pass
+
+
+def monitor_batch_job(
```

Comment:
Because of the completely different return type I would suggest that `monitor_batch_job` is not overloaded and we instead have a separate `monitor_batch_statistical_job` function. Also the default sleep time could be much shorter for statistical batch because we don't have to make a ton of request every time.

On the other hand, `monitor_batch_analysis` seems similar enough that we can keep the same implementation.